LeaderboardBackend.Test/Features/Users/RecoverAccountTests.cs

```diff
@@ -0,0 +1,178 @@
+        await context.SaveChangesAsync();
+        HttpResponseMessage res = await _client.GetAsync(Routes.RecoverAccount(recovery1.Id));
+        res.Should().HaveStatusCode(HttpStatusCode.NotFound);
+    }
+
+    public async Task TestRecovery_Used()
+    {
+        _clock.Reset(Instant.FromUnixTimeSeconds(10));
+        ApplicationContext context = _scope.ServiceProvider.GetRequiredService<ApplicationContext>();
+
+        AccountRecovery recovery = new()
+        {
+            CreatedAt = Instant.FromUnixTimeSeconds(0),
+            ExpiresAt = Instant.FromUnixTimeSeconds(0).Plus(Duration.FromHours(1)),
+            UsedAt = Instant.FromUnixTimeSeconds(5),
+            User = new()
+            {
+                Email = "test@email.com",
+                Password = "password",
+                Username = "username",
+                Role = UserRole.Confirmed
+            }
+        };
+
+        await context.AccountRecoveries.AddAsync(recovery);
+        await context.SaveChangesAsync();
+        HttpResponseMessage res = await _client.GetAsync(Routes.RecoverAccount(recovery.Id));
+        res.Should().HaveStatusCode(HttpStatusCode.NotFound);
+    }
+
+    public async Task TestRecovery_BannedUser()
+    {
+        _clock.Reset(Instant.FromUnixTimeSeconds(1));
+        ApplicationContext context = _scope.ServiceProvider.GetRequiredService<ApplicationContext>();
+
+        AccountRecovery recovery = new()
+        {
+            CreatedAt = Instant.FromUnixTimeSeconds(0),
+            ExpiresAt = Instant.FromUnixTimeSeconds(0).Plus(Duration.FromHours(1)),
+            User = new()
+            {
+                Email = "test@email.com",
+                Password = "password",
+                Username = "username",
+                Role = UserRole.Banned
+            }
+        };
+
+        await context.AccountRecoveries.AddAsync(recovery);
+        await context.SaveChangesAsync();
+        HttpResponseMessage res = await _client.GetAsync(Routes.RecoverAccount(recovery.Id));
+        res.Should().HaveStatusCode(HttpStatusCode.NotFound);
+    }
+
+    public async Task TestRecovery_Success()
+    {
+        _clock.Reset(Instant.FromUnixTimeSeconds(1));
+        ApplicationContext context = _scope.ServiceProvider.GetRequiredService<ApplicationContext>();
+
+        AccountRecovery recovery = new()
+        {
+            CreatedAt = Instant.FromUnixTimeSeconds(0),
+            ExpiresAt = Instant.FromUnixTimeSeconds(0).Plus(Duration.FromHours(1)),
+            User = new()
+            {
+                Email = "test@email.com",
+                Password = "password",
+                Username = "username",
+                Role = UserRole.Confirmed
+            }
+        };
+
+        await context.AccountRecoveries.AddAsync(recovery);
+        await context.SaveChangesAsync();
+        HttpResponseMessage res = await _client.GetAsync(Routes.RecoverAccount(recovery.Id));
+        res.Should().HaveStatusCode(HttpStatusCode.OK);
+    }
```

Comment:
Might wanna assert that the recovery entity was not modified.

Easily achievable with `newlyQueriedRecovery.Should().BeEquivalentTo(originalRecovery)`

Reply:
I mean shouldn't that be true for not just every test case but every GET request? In general, an HTTP GET request should never modify state.